packages/ui-components/src/components/Modal/ImportModal.js

```diff
@@ -0,0 +1,196 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import Typography from '@material-ui/core/Typography';
+import PropTypes from 'prop-types';
+import React, { useCallback, useState } from 'react';
+import styled from 'styled-components';
+
+import { SmallAlert } from '../Alert';
+import { Button, OutlinedButton } from '../Button';
+import { Dialog, DialogContent, DialogFooter, DialogHeader } from '../Dialog';
+import { FileUploadField } from '../Inputs';
+import { LoadingContainer } from '../LoadingContainer';
+
+const STATUS = {
+  IDLE: 'idle',
+  LOADING: 'loading',
+  SUCCESS: 'success',
+  ERROR: 'error',
+};
+
+const NO_FILE_MESSAGE = 'No file chosen';
+
+const Content = styled(DialogContent)`
+  text-align: left;
+  min-height: 220px;
+`;
+
+const Heading = styled(Typography)`
+  margin-bottom: 18px;
+`;
+
+export const ImportModal = ({
```

Comment:
Heavily inspired by [admin-panel/src/importExport/ImportModal.js](https://github.com/beyondessential/tupaia/blob/dev/packages/admin-panel/src/importExport/ImportModal.js). We could refactor that component to use the one introduced here at some point.

Also, ideally I should add `storybook` stories for this new component. I skipped that because I am not very familiar with `storybook`, but hopefully I'll find some time to change this in the near future 🤠